src/packages/@ncigdc/modern_components/Counts/Count.js

```diff
@@ -1,29 +1,57 @@
 // @flow
 import React from 'react';
-import { get } from 'lodash';
+import {
+  get,
+  isEqual,
+} from 'lodash';
 import { compose, withProps } from 'recompose';
 import withPropsOnChange from '@ncigdc/utils/withPropsOnChange';
 import GreyBox from '@ncigdc/uikit/GreyBox';
 
 export default compose(
   withProps(({ getter, path }) => ({
-    getCount: getter ? getter : v => get(v, path, ''),
+    getCount: getter || (v => get(v, path, '')),
   })),
-  withProps(({ viewer, path, loading, getCount }) => {
-    return {
-      count: loading ? '' : getCount(viewer),
-    };
-  }),
-  withPropsOnChange(['viewer'], ({ count, handleCountChange, loading }) => {
-    if (!loading && handleCountChange) {
-      handleCountChange(count);
-    }
-  }),
-)(({ count, children, loading, style }) => {
+  withProps(({
+    getCount,
+    loading,
+    viewer,
+  }) => ({
+    count: loading ? '' : getCount(viewer),
+  })),
+  withPropsOnChange(
+    (
+      {
+        viewer,
+      },
+      {
+        viewer: nextViewer,
+      },
+    ) => !(
+      isEqual(viewer, nextViewer)
```

Comment:
`withPropsOnChange` makes a shallow comparison of objects. This new approach avoids unnecessary renders.